subworkflows/nf-core/vcf_phase_shapeit5/main.nf

```diff
@@ -0,0 +1,81 @@
+include { BEDTOOLS_MAKEWINDOWS              } from '../../../modules/nf-core/bedtools/makewindows/main.nf'
+include { SHAPEIT5_PHASECOMMON              } from '../../../modules/nf-core/shapeit5/phasecommon/main'
+include { SHAPEIT5_LIGATE                   } from '../../../modules/nf-core/shapeit5/ligate/main'
+include { BCFTOOLS_INDEX as VCF_INDEX1      } from '../../../modules/nf-core/bcftools/index/main.nf'
+include { BCFTOOLS_INDEX as VCF_INDEX2      } from '../../../modules/nf-core/bcftools/index/main.nf'
+
+workflow VCF_PHASE_SHAPEIT5 {
+
+    take:
+    ch_vcf        // channel (mandatory): [ [id, ref], vcf, csi, pedigree, region ]
+    ch_ref        // channel (optional) : [ meta, ref, csi ]
+    ch_scaffold   // channel (optional) : [ meta, scaffold, csi ]
+    ch_map        // channel (optional) : [ meta, map]
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    // It is needed to generate a file containing the region to phase in a Chr \tab Start \tab End format
+    // The meta map needing to be conserved the following steps a required
+
+    // Keep the meta map and the region in two separated channel but keed id field to link them back
+    ch_vcf
+        .multiMap { m, vcf, csi, pedigree, region ->
+            metadata: [m.id, m]
+            region: [m.id,region ]
+        }
+        .set { ch_region }
+
+    // Create the File in bed format and use the meta id for the file name
+    ch_region.region
+        .collectFile { mid, region -> [ "${mid}.bed", region.replace(":","\t").replace("-","\t") ] }
+        .map { file -> [ file.simpleName, file ] }
```

Comment:
```suggestion
        .map { file -> [ file.baseName, file ] }
```
`simpleName` will remove all extensions from the file name. If the `meta.id` were to have dots in it, all characters after these dots would be removed too. `baseName` only removes the last extension (`.bed` in this case), see [here](https://www.nextflow.io/docs/latest/script.html#check-file-attributes) for more info

Reply:
Thanks !